backend/src/main/java/com/woowacourse/friendogly/club/domain/Club.java

```diff
@@ -158,11 +162,23 @@ public void addClubPet(List<Pet> pets) {
     }
 
     private void validateParticipatePet(Pet pet) {
-        if (!allowedGenders.contains(pet.getGender()) || !allowedSizes.contains(pet.getSizeType())) {
+        if (canNotJoin(pet)) {
             throw new FriendoglyException("모임에 데려갈 수 없는 강아지가 있습니다.");
         }
     }
 
+    private boolean canNotJoin(Pet pet) {
+        return !allowedGenders.contains(pet.getGender()) || !allowedSizes.contains(pet.getSizeType());
+    }
+
+    public boolean isJoinable(Member member, List<Pet> pets) {
+        boolean hasJoinablePet = pets.stream()
+                .anyMatch(pet -> !canNotJoin(pet));
+        boolean isNotFull = !this.memberCapacity.isCapacityReached(countClubMember());
+
+        return hasJoinablePet && isNotFull && isAlreadyJoined(member) && isOpen();
+    }
```

Comment:
```suggestion
    public boolean isJoinable(Member member, List<Pet> pets) {
        boolean hasJoinablePet = pets.stream()
                .anyMatch(pet -> canJoin(pet));
        boolean isNotFull = !this.memberCapacity.isCapacityReached(countClubMember());

        return hasJoinablePet && isNotFull && isAlreadyJoined(member) && isOpen();
    }
```

아까 리뷰 반영하신다면 이렇게 쓸 수 있어서 더 깔끔할 것 같아요!